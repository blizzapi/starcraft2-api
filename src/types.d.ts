import {
  BlizzAPIOptions,
  RegionIdAsNumberOrString,
  Sc2Realm,
} from 'blizzapi';

export interface StarCraft2APIOptions extends BlizzAPIOptions {}

export interface League {
  seasonId: number | string;
  queueId: number | string;
  teamType: number | string;
  leagueId: number | string;
}

export interface PlayerObject {
  regionId: RegionIdAsNumberOrString;
  realmId: Sc2Realm;
  profileId: number | string;
  locale?: string;
}
