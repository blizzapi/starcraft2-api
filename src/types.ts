import {
  BlizzAPIOptions,
  RegionIdAsNumberOrString,
  Sc2RealmAsNumberOrString,
} from "blizzapi";

export type StarCraft2APIOptions = BlizzAPIOptions;

export interface League {
  seasonId: number | string;
  queueId: number | string;
  teamType: number | string;
  leagueId: number | string;
}

export interface PlayerObject {
  regionId: RegionIdAsNumberOrString;
  realmId: Sc2RealmAsNumberOrString;
  profileId: number | string;
  locale?: string;
}
