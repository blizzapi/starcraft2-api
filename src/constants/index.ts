export { profileUrlLocales as PROFILE_URL_LOCALES } from "./profileUrlLocales";
