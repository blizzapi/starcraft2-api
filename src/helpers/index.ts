export { checkIfProfileIdLooksValid } from "./profileIds";
export {
  checkIfProfileUrlLocaleLooksValid,
  getAllProfileUrlLocales,
  profileUrlLocaleRegex,
  validateProfileUrlLocale,
} from "./profileUrlLocales";
export {
  checkIfProfileUrlLooksValid,
  constructProfileUrl,
  profileUrlRegex,
  unpackProfileUrl,
  validateProfileUrl,
} from "./profileUrls";
