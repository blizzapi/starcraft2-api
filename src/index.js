const Profile = require('./api/profile');
const Ladder = require('./api/ladder');
const Account = require('./api/account');
const Legacy = require('./api/legacy');

class StarCraft2API {
    constructor(clientID, clientSecret) {
        this.clientID = clientID;
        this.clientSecret = clientSecret;
    }
    
    Profile() {
        return new Profile(this.clientID, this.clientSecret);
    }

    Ladder() {
        return new Ladder(this.clientID, this.clientSecret);
    }

    Account() {
        return new Account(this.clientID, this.clientSecret);
    }

    Legacy() {
        return new Legacy(this.clientID, this.clientSecret);
    }
}

module.exports = StarCraft2API;
