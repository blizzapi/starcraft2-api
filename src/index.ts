import { BlizzAPI, BlizzUtils } from 'blizzapi';
import { QueryOptions } from 'blizzapi/dist/lib/interfaces';
import {
  ClientSecret,
  AccessToken,
  ClientId,
  RegionIdOrName,
  RegionIdAsNumberOrString,
  Locale,
  Sc2Realm,
} from 'blizzapi/dist/lib/types';

// tslint:disable no-class no-expression-statement no-this
export default class StarCraft2API extends BlizzAPI {
  constructor(
    region: RegionIdOrName,
    clientId: ClientId,
    clientSecret: ClientSecret,
    accessToken?: AccessToken,
    options?: QueryOptions,
  ) {
    super(region, clientId, clientSecret, accessToken, options);
  }

  queryStaticProfileData(regionId: RegionIdAsNumberOrString, locale?: Locale): Promise<object> {
    const queryLocale = locale || BlizzUtils.getDefaultLocaleNameForRegionId(regionId);
    return this.query(`/sc2/static/profile/${regionId}?locale=${queryLocale}`);
  }

  queryProfileMetadata(
    regionId: RegionIdAsNumberOrString,
    realmId: Sc2Realm,
    profileId: number | string,
    locale?: Locale,
  ): Promise<object> {
    const queryLocale = locale || BlizzUtils.getDefaultLocaleNameForRegionId(regionId);
    return this.query(
      `/sc2/metadata/profile/${regionId}/${realmId}/${profileId}?locale=${queryLocale}`,
    );
  }

  queryProfile(
    regionId: RegionIdAsNumberOrString,
    realmId: Sc2Realm,
    profileId: number | string,
    locale?: Locale,
  ): Promise<object> {
    const queryLocale = locale || BlizzUtils.getDefaultLocaleNameForRegionId(regionId);
    return this.query(`/sc2/profile/${regionId}/${realmId}/${profileId}?locale=${queryLocale}`);
  }

  queryLadderSummary(
    regionId: RegionIdAsNumberOrString,
    realmId: Sc2Realm,
    profileId: number | string,
    locale?: Locale,
  ): Promise<object> {
    const queryLocale = locale || BlizzUtils.getDefaultLocaleNameForRegionId(regionId);
    return this.query(
      `/sc2/profile/${regionId}/${realmId}/${profileId}/ladder/summary?locale=${queryLocale}`,
    );
  }

  queryPlayerLadder(
    regionId: RegionIdAsNumberOrString,
    realmId: Sc2Realm,
    profileId: number | string,
    ladderId: number | string,
    locale?: Locale,
  ): Promise<object> {
    const queryLocale = locale || BlizzUtils.getDefaultLocaleNameForRegionId(regionId);
    return this.query(
      `/sc2/profile/${regionId}/${realmId}/${profileId}/ladder/${ladderId}?locale=${queryLocale}`,
    );
  }

  queryGrandmasterLeaderboard(regionId: RegionIdAsNumberOrString): Promise<object> {
    return this.query(`/sc2/ladder/grandmaster/${regionId}`);
  }

  querySeason(regionId: RegionIdAsNumberOrString): Promise<object> {
    return this.query(`/sc2/ladder/season/${regionId}`);
  }

  queryPlayerAccount(accountId: number | string): Promise<object> {
    return this.query(`/s2/player/${accountId}`);
  }

  queryLegacyProfile(
    regionId: RegionIdAsNumberOrString,
    realmId: Sc2Realm,
    profileId: number | string,
  ): Promise<object> {
    return this.query(`/sc2/legacy/profile/${regionId}/${realmId}/${profileId}`);
  }

  queryLegacyLadders(
    regionId: RegionIdAsNumberOrString,
    realmId: Sc2Realm,
    profileId: number | string,
  ): Promise<object> {
    return this.query(`/sc2/legacy/profile/${regionId}/${realmId}/${profileId}/ladders`);
  }

  queryLegacyMatchHistory(
    regionId: RegionIdAsNumberOrString,
    realmId: Sc2Realm,
    profileId: number | string,
  ): Promise<object> {
    return this.query(`/sc2/legacy/profile/${regionId}/${realmId}/${profileId}/matches`);
  }

  queryLegacyLadder(
    regionId: RegionIdAsNumberOrString,
    ladderId: number | string,
  ): Promise<object> {
    return this.query(`/sc2/legacy/ladder/${regionId}/${ladderId}`);
  }

  queryLegacyAchievements(regionId: RegionIdAsNumberOrString): Promise<object> {
    return this.query(`/sc2/legacy/data/achievements/${regionId}`);
  }

  queryLegacyRewards(regionId: RegionIdAsNumberOrString): Promise<object> {
    return this.query(`/sc2/legacy/data/rewards/${regionId}`);
  }
}
// tslint:enable no-class no-expression-statement
