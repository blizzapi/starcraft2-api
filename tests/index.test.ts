import { StarCraft2API } from '../src/index';

describe('StarCraft2API', () => {
  it('should be defined', () => {
    expect(StarCraft2API).toBeDefined();
  });
});
