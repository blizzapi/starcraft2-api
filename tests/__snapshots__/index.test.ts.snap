// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarCraft2API checkIfProfileUrlLocaleLooksValid should match snapshot for correct parameters 1`] = `
Array [
  true,
  true,
  true,
  true,
  true,
  true,
  true,
  true,
  true,
  true,
]
`;

exports[`StarCraft2API checkIfProfileUrlLocaleLooksValid should match snapshot for correct parameters 2`] = `
Array [
  true,
  true,
  true,
  true,
  true,
  true,
  true,
  true,
  true,
  true,
]
`;

exports[`StarCraft2API checkIfProfileUrlLocaleLooksValid should match snapshot for malformed locales as parameters 1`] = `
Array [
  false,
  false,
  false,
  false,
]
`;

exports[`StarCraft2API checkIfProfileUrlLocaleLooksValid should match snapshot for non-existent locales as parameters 1`] = `
Array [
  true,
  true,
  true,
]
`;

exports[`StarCraft2API constructProfileUrl should match snapshot for invalid player objects 1`] = `
Array [
  "",
  "",
  "",
]
`;

exports[`StarCraft2API constructProfileUrl should match snapshot for valid player objects 1`] = `
Array [
  "https://starcraft2.com/en-us/profile/1/1/123456",
  "https://starcraft2.com/en-us/profile/2/2/987654",
]
`;

exports[`StarCraft2API constructProfileUrl should match snapshot for valid player objects and locale set to en-us 1`] = `
Array [
  "https://starcraft2.com/en-us/profile/1/1/123456",
  "https://starcraft2.com/en-us/profile/2/2/987654",
]
`;

exports[`StarCraft2API getAllProfileUrlLocales should match snapshot 1`] = `
Array [
  "en-us",
  "es-mx",
  "pt-br",
  "de-de",
  "en-gb",
  "es-es",
  "fr-fr",
  "it-it",
  "pl-pl",
  "ru-ru",
]
`;

exports[`StarCraft2API profileUrlRegex should match snapshot 1`] = `/starcraft2\\\\\\.com\\\\/\\(\\[A-z\\]\\{2\\}-\\[A-z\\]\\{2\\}\\)\\\\/profile\\\\/\\(1\\|2\\|3\\|5\\{1\\}\\)\\\\/\\(\\[0-9\\]\\{1\\}\\)\\\\/\\(\\[0-9\\]\\*\\)/gi`;

exports[`StarCraft2API queryGrandmasterLeaderboard should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/ladder/grandmaster/1",
}
`;

exports[`StarCraft2API queryLadderSummary should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/profile/1/1/123/ladder/summary?locale=en_US",
}
`;

exports[`StarCraft2API queryLeagueData should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/data/sc2/league/22/201/0/1",
}
`;

exports[`StarCraft2API queryLegacyAchievements should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/legacy/data/achievements/1",
}
`;

exports[`StarCraft2API queryLegacyLadder should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/legacy/ladder/1/123",
}
`;

exports[`StarCraft2API queryLegacyLadders should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/legacy/profile/1/1/123/ladders",
}
`;

exports[`StarCraft2API queryLegacyMatchHistory should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/legacy/profile/1/1/123/matches",
}
`;

exports[`StarCraft2API queryLegacyProfile should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/legacy/profile/1/1/123",
}
`;

exports[`StarCraft2API queryLegacyRewards should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/legacy/data/rewards/1",
}
`;

exports[`StarCraft2API queryPlayerAccount should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/player/123",
}
`;

exports[`StarCraft2API queryPlayerLadder should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/profile/1/1/123/ladder/123?locale=en_US",
}
`;

exports[`StarCraft2API queryProfile should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/profile/1/1/123?locale=en_US",
}
`;

exports[`StarCraft2API queryProfileMetadata should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/metadata/profile/1/1/123?locale=en_US",
}
`;

exports[`StarCraft2API querySeason should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/ladder/season/1",
}
`;

exports[`StarCraft2API queryStaticProfileData should match snapshot 1`] = `
Object {
  "data": "sample_starcraft2api_data",
  "endpoint": "/sc2/static/profile/1?locale=en_US",
}
`;

exports[`StarCraft2API unpackProfileUrl should match snapshot for invalid urls 1`] = `
Array [
  Object {},
  Object {},
  Object {},
]
`;

exports[`StarCraft2API unpackProfileUrl should match snapshot for valid urls 1`] = `
Array [
  Object {
    "profileId": "1084304",
    "realmId": "1",
    "regionId": "1",
  },
  Object {
    "profileId": "1084304",
    "realmId": "1",
    "regionId": "1",
  },
  Object {
    "profileId": "1084304",
    "realmId": "1",
    "regionId": "1",
  },
]
`;

exports[`StarCraft2API unpackProfileUrl should match snapshot for valid urls and includeLocale set to true 1`] = `
Array [
  Object {
    "locale": "en-us",
    "profileId": "1084304",
    "realmId": "1",
    "regionId": "1",
  },
  Object {
    "locale": "pl-pl",
    "profileId": "1084304",
    "realmId": "1",
    "regionId": "1",
  },
  Object {
    "locale": "ru-ru",
    "profileId": "1084304",
    "realmId": "1",
    "regionId": "1",
  },
]
`;

exports[`StarCraft2API validateProfileId should match snapshot for invalid profile ids 1`] = `
Array [
  false,
  false,
  false,
]
`;

exports[`StarCraft2API validateProfileId should match snapshot for valid profile ids 1`] = `
Array [
  true,
  true,
  true,
]
`;

exports[`StarCraft2API validateProfileUrl should match snapshot for invalid urls 1`] = `
Array [
  false,
  false,
  false,
]
`;

exports[`StarCraft2API validateProfileUrl should match snapshot for valid urls 1`] = `
Array [
  true,
  false,
  true,
]
`;

exports[`StarCraft2API validateProfileUrlLocale should match snapshot for non-existent locales as parameters 1`] = `
Array [
  false,
  false,
  false,
]
`;
